### PR TITLE
Add marginal tests for Zero Mean Normal, and fix bugs in logp/logp_score

### PR DESCRIPTION
**Pull request by srvasude**
Add tests that numerically integrate to find logp and logp_score.

- Fix bug in logp and logp_score.
- Use log1p in Student-T calculation for better numerical stability.

### CODE REVIEW

File: cxx/distributions/zero_mean_normal.cc [hunk 1]
```diff
@@ -37,20 +37,20 @@ void ZeroMeanNormal::unincorporate(const double& x) {
 double ZeroMeanNormal::logp(const double& x) const {
   // Equation (119) of https://www.cs.ubc.ca/~murphyk/Papers/bayesGauss.pdf
   double alpha_n = alpha + N / 2.0;
-  double beta_n = beta + var * N;
+  double beta_n = beta + 0.5 * var * N;
```
Comment (ThomasColthurst) [inline cxx/distributions/zero_mean_normal.cc:40]: Oops!  Thanks for catching this.

File: cxx/distributions/zero_mean_normal.cc [hunk 2]
```diff
@@ -37,20 +37,20 @@ void ZeroMeanNormal::unincorporate(const double& x) {
...
   double t_variance = beta_n / alpha_n;
   return log_t_distribution(x, 2.0 * alpha_n, t_variance);
 }
 
 double ZeroMeanNormal::logp_score() const {
   // Marginal likelihood from Page 10 of
```
Comment (ThomasColthurst) [inline cxx/distributions/zero_mean_normal.cc:46]: The Page # needs updating.

File: cxx/distributions/zero_mean_normal.cc [hunk 3]
```diff
@@ -37,20 +37,20 @@ void ZeroMeanNormal::unincorporate(const double& x) {
...
-  // https://j-zin.github.io/files/Conjugate_Bayesian_analysis_of_common_distributions.pdf
+  // https://people.eecs.berkeley.edu/~jordan/courses/260-spring10/lectures/lecture5.pdf
```
Comment (ThomasColthurst) [inline cxx/distributions/zero_mean_normal.cc:47]: Is there a reason you like this reference better?  I prefer the old one because it has the explicit formula for the marginal likelihood of the data.

**Review by ThomasColthurst — APPROVED** [review]
Any idea why the pre-commit.ci check is failing (for this and other pull requests)?

File: cxx/distributions/zero_mean_normal.cc (hunk 3 above)
Comment (srvasude) [reply cxx/distributions/zero_mean_normal.cc:47]: It's incorrect because it has a factor of 2 on the beta (caught by the marginalization tests).
